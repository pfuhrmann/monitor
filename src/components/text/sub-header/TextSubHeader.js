import React from 'react';
import PropTypes from 'prop-types';
import './TextSubHeader.scss';

const TextSubHeader = ({ children, className }) => (
  <h3 className={className}>{children}</h3>
);

TextSubHeader.propTypes = {
  children: PropTypes.node,
};

TextSubHeader.defaultProps = {
  children: '',
};

export default TextSubHeader;
