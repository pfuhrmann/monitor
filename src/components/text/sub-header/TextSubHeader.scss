@import "../../../variables.scss";

h3 {
  font-size: 1rem;
  font-weight: $font-weight-semi-bold;
}
