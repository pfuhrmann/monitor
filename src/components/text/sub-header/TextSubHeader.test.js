import React from 'react';
import ReactDOM from 'react-dom';
import TextSubHeader from './TextSubHeader';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextSubHeader />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly h3 tag with text', () => {
  const component = renderer.create(
    <TextSubHeader>Header</TextSubHeader>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with custom class', () => {
  const component = renderer.create(
    <TextSubHeader className="test-class">Header</TextSubHeader>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
