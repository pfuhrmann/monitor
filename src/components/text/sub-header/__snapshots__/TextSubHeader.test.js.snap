// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly h3 tag with text 1`] = `
<h3>
  Header
</h3>
`;

exports[`renders correctly with custom class 1`] = `
<h3
  className="test-class"
>
  Header
</h3>
`;
